### ISSUE TITLE
please Expose redis process path

### ISSUE THREAD

**Issue by n0mer** [kstyrc/embedded-redis#32]
currently in redis.embedded.util.JarUtil#extractExecutableFromJar()
```java
import com.google.common.io.Files;
//skip
public class JarUtil {

    public static File extractExecutableFromJar(String executable) throws IOException {
        File tmpDir = Files.createTempDir();
```

Then in redis.embedded.RedisServer
```java
    public RedisServer(Integer port) throws IOException {
        super(port);
        File executable = JarUtil.extractExecutableFromJar(RedisRunScriptEnum.getRedisRunScript());
```

then in redis.embedded.AbstractRedisInstance
```java
    public synchronized void start() throws EmbeddedRedisException {
        if (active) {
            throw new EmbeddedRedisException("This redis server instance is already running...");
        }
        try {
            redisProcess = createRedisProcessBuilder().start();
```

and

```java
    private ProcessBuilder createRedisProcessBuilder() {
        File executable = new File(args.get(0));
        ProcessBuilder pb = new ProcessBuilder(args);
        pb.directory(executable.getParentFile());
        return pb;
    }
```

So, as you can see, there is no way to read "executable" path outside of AbstractRedisInstance.

Is it possible to expose "executable" via getter at AbstractRedisInstance?

Sometimes (at least on windows) redis-server-64.exe is still running. I'm going to at least print the path of the executable to log, to be able to see later whether this subdirectory inside "java.io.tmpdir" was actually removed or not on exit.

**Comment by kstyrc** [comment]
Of course it should. However, maybe we should rather do proper cleanup of the redis process after cleanup? Under what circumstances the redis stays up after execution?

**Comment by n0mer** [comment]
i use Redis embedded in integration tests. spring context is managed by spring-boot, so redis cfg is the following:

```java
    @Bean(destroyMethod = "stop")
    RedisServer redisServer() {

//...
        try {
            RedisServer redisServer = new RedisServer(port);
            redisServer.start();
            return redisServer;
        } catch (IOException e) {
//...
        }
    }
}
```

There is another c-tor for RedisServer that allows to specify the File for redis executable. But again, how can we specify that if there is no way to obtain tmp.path from previous invocation :)

On windows desktop (jdk-8u25, win7 64bit), in sysinternal's process explorer, sometimes child redis-server.exe process (started from java process) stays alive even when it's former parent process is already finished. So, to avoid the exception or "redis alredy started", you need to always check manually in process explorer whether any of already running redis-server instances exist.

Even more - exception that is thrown when redis is already running, is nothing but informative.

**Comment by n0mer** [comment]
In other words, and to put it short:
* it would be beneficial to provide additional constructor, that will accept "tmp.dir" as a parameter. Then, if you have this directory somewhere inside gradle's ./build, then you'll notice that during "clean" phase something is still running :)
* if redis-server is embedded, then process with "redis-embed" prefix with PPID pointing nowhere is a sign of something went wrong; so this fact can be at least WARN'ed in log during embedded server startup.

**Comment by kstyrc** [comment]
Think it's doable and I'll address that in the next release. Would you like to make contribution? Apart of that, what is the reason redis process keeps running? It should be wiped out on JVM exit. Maybe we should treat it as a bug & fix rather than expose lib internals?

Thanks for the input.

**Comment by n0mer** [comment]
Sorry, this is what i know.
Since then i removed redis-embedded from the project and not using it, because of this hassle of running integration tests on dev machine.

There is no way to workaround this process issue in build scripts, because tmp folder name is neither exposed nor explicitly logged. So, quickest way, for now, is to expose folder directory. And then dig & troubleshoot every possible combination of causes why the process still runs even when there is no parent any more.

**Comment by kstyrc** [comment]
As you can see, on the main page of the project, we currently have CI running for Windows on AppVeyor. It truly fails on Windows. My guess is that redis process does not exit correctly and/or stales and takes up port. Will try to address & fix it soon.

**Comment by kstyrc** [comment]
Will try to fix embedded-redis subsequent runs in #40. If not, then I'll expose redis process path.
